detectron2/modeling/meta_arch/multi_modal_rcnn.py

```diff
@@ -0,0 +1,255 @@
+            pixel_mean, pixel_std: list or tuple with #channels element, representing
+                the per-channel mean and std to be used to normalize the input image
+            input_format: describe the meaning of channels of input. Needed by visualization
+            vis_period: the period to run visualization. Set to 0 to disable.
+            feature_dropout_rate: the rate to dropout backbone features
+            feature_noise_std: the standard deviation of the noise to add to backbone features
+            min_size_input: the minimum size of the input image
+        """
+
+        super().__init__(backbone=backbone, proposal_generator=proposal_generator, roi_heads=roi_heads, pixel_mean=pixel_mean, pixel_std=pixel_std, input_format=input_format, vis_period=vis_period)
+
+        # Compute the expected output shape for the text encoder
+        backbone_shapes = self.backbone.output_shape()
+        channels = backbone_shapes[self.backbone._out_features[-1]].channels
+        largest_stride = backbone_shapes[self.backbone._out_features[-1]].stride
+        smallest_feature_size = (self.backbone._square_pad // largest_stride if self.backbone._square_pad else PROCESSED_IMAGE_SIZE[0] // largest_stride,
+                                self.backbone._square_pad // largest_stride if self.backbone._square_pad else PROCESSED_IMAGE_SIZE[1] // largest_stride)
+
+        self.feature_dropout = nn.Dropout(feature_dropout_rate)
+        self.feature_noise_std = feature_noise_std
+        # TODO: TextFeatureExtractor as input argument
+        self.text_encoder = TextFeatureExtractor(
+            output_shape=384,
+            expected_output_shape=(channels, int(smallest_feature_size[0]), int(smallest_feature_size[1])), 
+            fpn_keys=self.backbone._out_features,
+            device="cuda")
+
+    def forward(self, batched_inputs: list[dict[str, torch.Tensor]]):
+        """Forward pass for the model.
+
+        Args:
+            batched_inputs (list[dict]): A list of dictionaries, each containing an image and its corresponding text.
+
+        Returns:
+            dict: A dictionary containing the losses.
+        """
+
+        if not self.training:
+            return self.inference(batched_inputs)
+
+        images = self.preprocess_image(batched_inputs)
+        if "instances" in batched_inputs[0]:
+            gt_instances = [x["instances"].to(self.device) for x in batched_inputs]
+        else:
+            gt_instances = None
+
+        features = self.backbone(images.tensor)
+
+        feature_shapes = {k: v.shape for k, v in features.items()}
+        text_features = self.text_encoder([batched_input['text'] for batched_input in batched_inputs], feature_shapes)
+
+        for k in features:
+            # Dropout features
+            features[k] = self.feature_dropout(features[k])
+
+            # Add noise
+            features[k] = features[k] + torch.randn_like(features[k]) * self.feature_noise_std
```

Comment:
Could you please add some extra comments on why you do the feature dropout and add the noise?